bus-mapping/src/circuit_input_builder/execution.rs

```diff
@@ -320,6 +320,8 @@ impl_expr!(CopyDataType, u64::from);
 pub struct CopyStep {
     /// Byte value copied in this step.
     pub value: u8,
+    /// Byte value before this step.
+    pub prev_value: Option<u8>,
```

Comment:
This is the most logical approach: everywhere there is a `value`, there is a `value_prev` too.

Reply:
We could say that in a reader step, `value_prev = value`. That would simplify code without the `Option` wrapper, and always handling reader and writer the same way.

Reply:
currently only write step has `value_prev`, if do the same for reader step, calculating  additinal `value_read_prev_rlc` making no sense since it is equal to value_word_rlc. wdyk?

Reply:
Sure then the reader case is different as an optimisation.